src/buf/buf.rs

```diff
@@ -1,11 +1,32 @@
-use super::{IntoBuf, Take, Reader, FromBuf, Chain};
+use super::{Chain, FromBuf, IntoBuf, Reader, Take};
 use byteorder::{BigEndian, ByteOrder, LittleEndian};
 
 use std::{cmp, io::IoSlice, ptr};
 
+/// An error which occurred while attempting
+/// to get a value from a [`Buf`](trait.Buf.html).
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum TryGetError {
+    /// Indicates that there were not enough remaining
+    /// bytes in the buffer to read a value.
+    NotEnoughBytes,
+}
+
+impl std::fmt::Display for TryGetError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
+        match self {
+            TryGetError::NotEnoughBytes => write!(f, "Not enough bytes in buffer to read value")?,
+        }
+
+        Ok(())
```

Comment:
This would eat the error that my be returned by `write!`
```suggestion
```

Reply:
Well, there is the `?` operator after `write!` - I would expect that an error would be returned there. I guess it would be cleaner just return the result directly from `write!`, however.

Reply:
ha, well, I missed it, sorry for the noise.